motorlib/motor.py

```diff
@@ -182,167 +294,250 @@ def runSimulation(self, callback=None):
-            simRes.channels['pressure'].addData(pressure)
+            lastKn = simRes.channels["kn"].getLast()
+            pressure: float = self.calcIdealPressure(perGrainReg, dThroat, lastKn)
+            simRes.channels["pressure"].addData(pressure)
 
             # Calculate Mach Number
-            perGrainMachNumber = [0 for grain in self.grains]
+            perGrainMachNumber = [0] * len(self.grains)
             for gid, grain in enumerate(self.grains):
-                perGrainMachNumber[gid] = self.calcMachNumber(pressure, perGrainMassFlux[gid])
-            simRes.channels['machNumber'].addData(perGrainMachNumber)
+                perGrainMachNumber[gid] = self.calcMachNumber(
+                    pressure, perGrainMassFlux[gid]
+                )
+            simRes.channels["machNumber"].addData(perGrainMachNumber)
 
             # Calculate Exit Pressure
             _, _, gamma, _, _ = self.propellant.getCombustionProperties(pressure)
             exitPressure = self.nozzle.getExitPressure(gamma, pressure)
-            simRes.channels['exitPressure'].addData(exitPressure)
+            simRes.channels["exitPressure"].addData(exitPressure)
 
             # Calculate force
-            force = self.calcForce(simRes.channels['pressure'].getLast(), dThroat, exitPressure)
-            simRes.channels['force'].addData(force)
+            force = self.calcForce(
+                simRes.channels["pressure"].getLast(), dThroat, exitPressure
+            )
+            simRes.channels["force"].addData(force)
+            simRes.channels["time"].addData(simRes.channels["time"].getLast() + dTime)
 
-            simRes.channels['time'].addData(simRes.channels['time'].getLast() + dTime)
+            slagRate: float = self.getSlagRate(pressure)
 
-            # Calculate any slag deposition or erosion of the throat
-            if pressure == 0:
-                slagRate = 0
-            else:
-                slagRate = (1 / pressure) * self.nozzle.getProperty('slagCoeff')
-            erosionRate = pressure * self.nozzle.getProperty('erosionCoeff')
+            erosionRate = pressure * self.nozzle.getProperty("erosionCoeff")
             change = dTime * ((-2 * slagRate) + (2 * erosionRate))
-            simRes.channels['dThroat'].addData(dThroat + change)
+            simRes.channels["dThroat"].addData(dThroat + change)
 
             if callback is not None:
                 # Uses the grain with the largest percentage of its web left
-                progress = max([g.getWebLeft(r) / g.getWebLeft(0) for g, r in zip(self.grains, perGrainReg)])
-                if callback(1 - progress): # If the callback returns true, it is time to cancel
+                progress = max(
+                    [
+                        g.getWebLeft(r) / g.getWebLeft(0)
+                        for g, r in zip(self.grains, perGrainReg)
+                    ]
+                )
+                if callback(
```

Comment:
I like a lot of what ruff has done, but this just reads worse to me. Absolutely no reason to split a line this short with one function call with one arg over so many lines... I'll look into if there's a way to configure ruff.

Reply:
got it, reverted